### ISSUE TITLE
Profiles missing on relaunch

### ISSUE THREAD

**Issue by yamahito** [MultiMC/Launcher#4462]
### Operating System

macOS

### Description of bug

Previously authenticated profiles should be available on (re)starting the loader, but are missing.

### Steps to reproduce

- open MMC
- Run instance
- realise that the (mojang) profile is missing, re-enter credentials
- forget about it until next time

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by yamahito** [comment]
<img width="532" alt="Screenshot 2022-01-28 at 19 08 55" src="https://user-images.githubusercontent.com/3420242/151606827-89490f7a-c443-4f3a-acde-4fe12eb6b6ed.png">

**Comment by yamahito** [comment]
Except that the Mojang account in question has already been added a couple of times.

**Comment by yamahito** [comment]
Can confirm it happens on restart of MultiMC: file permissions or some annoying mac security 'feature'?  Looking into it now.

Instances are saved.

**Comment by yamahito** [comment]
Giving MultiMC "full disk access" in the control panel security settings is not a workaround, sadly.

**Comment by yamahito** [comment]
Is there anything I can do, or information I could provide, which would aid diagnosis?  I'm assuming that other macOS users aren't generally having this problem, so reproducing the bug likely needs some further analysis.  Happy to help, but could do with someone suggesting a starting point!

**Comment by phit** [comment]
open the MultiMC-0.log in the apps directory and check for errors saving files could be a starting point

**Comment by phit** [comment]
This issue is being closed for inactivity. If you're still having trouble with this, you can still reply to this issue and I'll try to help or feel free to come by our [Discord server](https://discord.gg/multimc) and ask for help there.

**Comment by yamahito** [comment]
Sorry this took so long to get around to.  Work and family life come first (and fast)!

Turns out the problem was with permissions in the `/Applications/MultiMC.app/Data/profiles.json` file, which was set to be owned by another local user (?) and with 600 permissions (?!).  Not sure how that happened - pretty sure it's not something that I did manually, and no other files in the directory were affected.

Anyway, fixing the permissions fixed the problems.  Happy for you to close, since whatever happened seems to be unique to me!
